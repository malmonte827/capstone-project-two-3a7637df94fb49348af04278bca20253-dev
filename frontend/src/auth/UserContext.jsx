import React from "react"

/** COntext provides current user object throughout app */

const UserContext = React.createContext()

export default UserContext