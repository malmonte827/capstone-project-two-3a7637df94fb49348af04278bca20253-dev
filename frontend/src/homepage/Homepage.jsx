import React, { useContext } from "react";
import { Link, useLocation } from "react-router-dom";
import UserContext from "../auth/UserContext";
import Alert from "../common/Alert"
import "./Homepage.css"

/** Homepage
 *
 * If the user is logged in, it greets them by their first name or username.
 * If the user is not logged in, it shows login and signup buttons.
 * 
 * Routes -> Homepage
 *
 * Routed at /
 */
function Homepage() {
    const { currentUser } = useContext(UserContext);

    const location = useLocation();
    const successMessage = location.state?.message || null;

    return (
        <div className="Homepage">
            <div className="container text-center">
                <h1 className="mb-4 fw-bold">Capstone</h1>
                <p className="lead">
                    This is my Capstone project hope you like it!
                </p>
                {currentUser ? (
                    <h2>
                        Welcome Back,{" "}
                        {currentUser.firstName || currentUser.username}
                    </h2>
                ) : (
                    <p>
                        <Link
                            className="btn btn-primary fw-bold me-3"
                            to="/login"
                        >
                            Login
                        </Link>
                        <Link
                            className="btn btn-primary fw-bold me-3"
                            to="/signup"
                        >
                            Signup
                        </Link>
                    </p>
                )}

            <div className="col-lg-4 offset-lg-4">
            {successMessage
              ?
              <Alert type="success" messages={[successMessage]} />
              : null}
            </div>
            </div>
        </div>
    );
}
export default Homepage
