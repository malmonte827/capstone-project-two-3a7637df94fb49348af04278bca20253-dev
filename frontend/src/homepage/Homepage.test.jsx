import React from "react";
import Homepage from "./Homepage";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import UserContext from "../auth/UserContext";

describe("<Homepage />", () => {
    const currentUser = { firstName: "John", username: "john_doe" };

    it("renders without crashing", () => {
        render(
            <MemoryRouter>
                <UserContext.Provider value={{ currentUser }}>
                    <Homepage />
                </UserContext.Provider>
            </MemoryRouter>
        );
    });

    it("matches snapshot", () => {
        const { asFragment } = render(
            <MemoryRouter>
                <UserContext.Provider value={{ currentUser }}>
                    <Homepage />
                </UserContext.Provider>
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    describe("Displays correct user information", () => {
        it("greets the logged-in user", () => {
            render(
                <MemoryRouter>
                    <UserContext.Provider value={{ currentUser }}>
                        <Homepage />
                    </UserContext.Provider>
                </MemoryRouter>
            );

            expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent(
                "Welcome Back, John"
            );
        });

        it("shows login and signup buttons if no user is logged in", () => {
            render(
                <MemoryRouter>
                    <UserContext.Provider value={{ currentUser: null }}>
                        <Homepage />
                    </UserContext.Provider>
                </MemoryRouter>
            );

            expect(
                screen.getByRole("link", { name: /Login/i })
            ).toBeInTheDocument();
            expect(
                screen.getByRole("link", { name: /Signup/i })
            ).toBeInTheDocument();
        });
    });

    describe("Displays success message", () => {
        it("shows success message if passed in location state", () => {
            const successMessage = "Your action was successful!";
            render(
                <MemoryRouter
                    initialEntries={[{ state: { message: successMessage } }]}
                >
                    <UserContext.Provider value={{ currentUser }}>
                        <Homepage />
                    </UserContext.Provider>
                </MemoryRouter>
            );

            expect(screen.getByText(successMessage)).toBeInTheDocument();
        });

        it("does not show success message if no message is passed", () => {
            render(
                <MemoryRouter>
                    <UserContext.Provider value={{ currentUser }}>
                        <Homepage />
                    </UserContext.Provider>
                </MemoryRouter>
            );

            expect(
                screen.queryByText(/Your action was successful!/)
            ).not.toBeInTheDocument();
        });
    });
});
