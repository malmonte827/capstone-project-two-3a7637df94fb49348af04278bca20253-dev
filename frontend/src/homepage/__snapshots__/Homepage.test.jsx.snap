// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Homepage /> > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <div
      class="container text-center"
    >
      <h1
        class="mb-4 fw-bold"
      >
        Capstone
      </h1>
      <p
        class="lead"
      >
        This is my Capstone project hope you like it!
      </p>
      <h2>
        Welcome Back, John
      </h2>
      <div
        class="col-lg-4 offset-lg-4"
      />
    </div>
  </div>
</DocumentFragment>
`;
